guibot/finder.py

```diff
@@ -128,7 +148,7 @@ def from_string(raw):
                 arg = False
             elif re.match(r"-?\d+$", arg):
                 arg = int(arg)
-            elif re.match(r"[\d.]+$", arg):
+            elif re.match(r"-?\d+(?:\.\d+)?$", arg):
```

Comment:
Thanks for extending this, it was pretty limited and only meant for floats encountered some years ago.